config/initializers/wicked_pdf.rb

```diff
@@ -3,3 +3,11 @@
   #:layout => "pdf.html",
   :exe_path => `bundle exec which wkhtmltopdf`.chomp
 }
+
+# A monkey-patch to remove WickedPdf's monkey-patch, as it clashes with ViewComponents.
+class WickedPdf
+  module PdfHelper
+    remove_method(:render)
+    remove_method(:render_to_string)
+  end
+end
```

Comment:
I let @Matt-Yorkley answer on this point ; he helps me a lot on this particular point... 🙏

Reply:
> It would be nice to mention where you found this

I wrote it. Basically _both_ `wicked_pdf` and `view_component` apply a slightly different monkey-patch on the `ActionController#render` and `ActionController#render_to_string` methods, but they're not compatible. I wrote an initial comment here: https://github.com/openfoodfoundation/openfoodnetwork/pull/7284#issuecomment-811211809

So this basically removes the `wicked_pdf` monkey-patch for those methods, which means when we're rendering a PDF we now need to explicitly use `#render_to_pdf_with_wicked_pdf` (for example).

Yeah it's a bit of a mess... :see_no_evil: 

Maybe we can add some more detailed comments to make the context a bit clearer.